common.mk

```diff
@@ -65,29 +65,30 @@ HELP_COMMANDS += \
 # include additional subproject make fragments
 # see HELP_COMPILATION_VARIABLES
 #########################################################################################
-include $(base_dir)/generators/cva6/cva6.mk
-include $(base_dir)/generators/ibex/ibex.mk
-include $(base_dir)/generators/ara/ara.mk
 include $(base_dir)/generators/tracegen/tracegen.mk
-include $(base_dir)/generators/nvdla/nvdla.mk
-include $(base_dir)/generators/radiance/radiance.mk
 include $(base_dir)/tools/torture.mk
+-include $(base_dir)/generators/cva6/cva6.mk
+-include $(base_dir)/generators/ibex/ibex.mk
+-include $(base_dir)/generators/ara/ara.mk
+-include $(base_dir)/generators/nvdla/nvdla.mk
+-include $(base_dir)/generators/radiance/radiance.mk
+
 
 #########################################################################################
 # Prerequisite lists
 #########################################################################################
 # Returns a list of files in directories $1 with single file extension $2.
 # If available, use 'fd' to find the list of files, which is faster than 'find'.
 ifeq ($(shell which fd 2> /dev/null),)
-	lookup_srcs = $(shell find -L $(1)/ -name target -prune -o \( -iname "*.$(2)" ! -iname ".*" \) -print 2> /dev/null)
+	lookup_srcs = $(shell find -L $(1)/ -name target -prune -o \( ! -xtype l -a -iname "*.$(2)" ! -iname ".*" \) -print 2> /dev/null)
 else
 	lookup_srcs = $(shell fd -L -t f -e $(2) . $(1))
 endif
 
 # Returns a list of files in directories $1 with *any* of the file extensions in $2
 lookup_srcs_by_multiple_type = $(foreach type,$(2),$(call lookup_srcs,$(1),$(type)))
 
-CHECK_SUBMODULES_COMMAND = echo "Checking all submodules in generators/ are initialized. Uninitialized submodules will be displayed" ; ! git submodule status $(base_dir)/generators | grep ^-
+CHECK_SUBMODULES_COMMAND = echo "Checking required submodules in generators/ are initialized. Uninitialized submodules will be displayed" ; ! git submodule status $(base_dir)/generators | grep '^-.*' | grep -vE "(ara|caliptra|compress)"
```

Comment:
This to me seems brittle, I expect users to forget to add things to this list. Can we auto-generate this list based on the build.sbt (I know this would be hacky but autogenerate this based on the build.sbt file)?

Reply:
This would need a "Single-source-of-truth" for listing what submodules are required/optional. I don't believe build.sbt sould be that single-source-of-truth... it would be another consumer

We could define a yaml/json somewhere in chipyard that all the scripts reference.

Reply:
Is the intent that this YAML/JSON feeds into the `build.sbt`, which then parses this file? If so, then we need to encode dependencies, which, in my opinion, is too heavy-handed. If it's separate then you need to update the YAML/JSON and the `build.sbt` which also leads to the same issue (de-sync from both sets of inputs).

Brainstorming: We have a `map[string -> SBTProjects]` in the `build.sbt`. Users must add their project to this `map`, then there is a required `list[string]` that says the default required submodules. This is then passed to the main Chipyard project? Then you just need to parse that `build.sbt` line (this would still be janky).

Reply:
So there are three options I see:

1. Heavy-handed approach, with new YAML/JSON that serves as single-source-of-truth that all the other scripts and build.sbt parse 
2. Use build.sbt as single-source-of-truth, require parsing build.sbt in various scripts... (very ugly)
3. No single-source-of-truth

I'm of the opinion that approach 1 makes the most sense, but makes the most engineering. Approach 3 is the easiest for now. I'm not a fan of the intermediate approach 2.

Reply:
FWIW I don't think (2) is that bad (bad but significantly less effort than 1): For example:

```scala
val projectNames: Seq[String] = Seq(
  "projectA",
  "projectB",
  "projectC",
  "projectD"
)

val projects = projectNames.map(name => project(name).settings(
  // Add common settings here, if needed.
  version := "1.0.0",
  scalaVersion := "2.13.8" // or your desired Scala version
))
```

With that being said, I vote for (3) for now, if people want to fix this later (when they have time - or get annoyed enough) then they can.